test/support/utils.ex

```diff
@@ -9,31 +9,50 @@ defmodule VintageNetWiFiTest.Utils do
 
   @spec udhcpc_child_spec(VintageNet.ifname(), String.t()) :: Supervisor.child_spec()
   def udhcpc_child_spec(ifname, hostname) do
+    beam_notify = Application.app_dir(:beam_notify, "priv/beam_notify")
+
     %{
       id: :udhcpc,
-      restart: :permanent,
-      shutdown: 500,
-      start:
-        {MuonTrap.Daemon, :start_link,
-         [
-           "udhcpc",
-           [
-             "-f",
-             "-i",
-             ifname,
-             "-x",
-             "hostname:#{hostname}",
-             "-s",
-             Application.app_dir(:vintage_net, ["priv", "udhcpc_handler"])
-           ],
-           [stderr_to_stdout: true, log_output: :debug, log_prefix: "udhcpc(#{ifname}): "]
-         ]},
-      type: :worker
+      start: {
+        VintageNet.Interface.IfupDaemon,
+        :start_link,
+        [
+          [
+            ifname: ifname,
+            command: "udhcpc",
+            args: [
+              "-f",
+              "-i",
+              ifname,
+              "-x",
+              "hostname:#{hostname}",
+              "-s",
+              beam_notify
+            ],
+            opts: [
+              stderr_to_stdout: true,
+              log_output: :debug,
+              log_prefix: "udhcpc(#{ifname}): ",
+              env: %{
+                "BEAM_NOTIFY" => beam_notify,
+                "BEAM_NOTIFY_OPTIONS" => "-p /tmp/beam_notify-89749473 -e"
+              }
```

Comment:
```suggestion
              env: BEAMNotify.env(name: "vintage_net_comm", report_env: true)
```

This is what I did over here: https://github.com/nerves-networking/vintage_net/blob/main/test/support/utils.ex#L9-L37